pkgs/by-name/mu/muffon/package.nix

```diff
@@ -0,0 +1,40 @@
+{ lib
+, fetchurl
+, appimageTools
+, makeWrapper
+}:
+
+let
+  pname = "muffon";
+  version = "2.0.3";
+  src = fetchurl {
+    url = "https://github.com/staniel359/${pname}/releases/download/v${version}/${pname}-${version}-linux-x86_64.AppImage";
```

Comment:
Don't re-use `pname`, use a string literal please.

Reply:
Like this? 
```nix
{
  lib,
  fetchurl,
  appimageTools,
  makeWrapper,
}:

let
  version = "2.0.3";
  src = fetchurl {
    url = "https://github.com/staniel359/muffon/releases/download/v${version}/muffon-${version}-linux-x86_64.AppImage";
    hash = "sha256-2eLe/xvdWcOcUSE0D+pMOcOYCfFVEyKO13LiaJiZgX0=";
  };
  appimageContents = appimageTools.extractType2 { pname = "muffon"; inherit src version; };
in
appimageTools.wrapType2 {
  pname = "muffon";
  inherit src version;

  extraInstallCommands = ''
    source "${makeWrapper}/nix-support/setup-hook"
    wrapProgram $out/bin/muffon \
      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
    install -m 444 -D ${appimageContents}/muffon.desktop -t $out/share/applications
    substituteInPlace $out/share/applications/muffon.desktop \
      --replace-fail 'Exec=AppRun' 'Exec=muffon'
    install -m 444 -D ${appimageContents}/muffon.png \
      $out/share/icons/hicolor/512x512/apps/muffon.png
  '';

  meta = {
    description = "Advanced multi-source music streaming client";
    homepage = "https://muffon.netlify.app/";
    changelog = "https://github.com/staniel359/muffon/releases/tag/v${version}";
    license = lib.licenses.agpl3Only;
    maintainers = with lib.maintainers; [ octodi ];
    mainProgram = "muffon";
    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
    platforms = [ "x86_64-linux" ];
  };
}```

Reply:
Yes, true. I forgot src should stay independent from package name, that'll work

Reply:
If you really want me to check what you have changed in the file, please just copy the changed files only, it will be easier and faster for me to check.

Reply:
Actually is there any good/reliable resources/documentation on where to reuse pname and where to type it out?

I see muffon/pname used in:
- src (where explicit is recommended)
- extraInstallCommands
- meta (changelog link and such)

With sources not using pname makes sense, and drupols comment only referred to sources as far as i can tell, but i do still wonder about the other two.

Reply:
@LordGrimmauld Could you please suggest changes in the new commit